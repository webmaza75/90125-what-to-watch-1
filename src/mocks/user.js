const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

export default user;
