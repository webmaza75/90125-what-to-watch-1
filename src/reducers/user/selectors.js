import NameSpace from '../name-spaces.js';

const NAME_SPACE = NameSpace.USER;

export const getUser = (state) => {
  return state[NAME_SPACE].userInfo;
};

export const getError = (state) => {
  return state[NAME_SPACE].error;
};

export const getValidationError = (state) => {
  return state[NAME_SPACE].validationError;
};

export const checkIsAuthorizedUser = (state) => {
  const user = state[NAME_SPACE].userInfo;
  return !!(user && user.id);
};

export const checkIsRequiredAuthorization = (state) => {
  return state[NAME_SPACE].isRequiredAuthorization;
};

export const checkIsLoadedUserInfo = (state) => {
  return state[NAME_SPACE].isLoadedUserInfo;
};
