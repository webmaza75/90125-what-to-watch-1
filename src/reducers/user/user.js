import ActionType from '../../actions/action-type.js';
import {ActionCreator} from '../../actions/actions.js';

const initialState = {
  userInfo: undefined,
  error: undefined,
  validationError: undefined
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SIGN_IN_USER:
      return {
        ...state,
        userInfo: action.payload
      };
    case ActionType.SIGN_IN_USER_ERROR:
      return {
        ...state,
        error: action.payload
      };
    case ActionType.VALIDATE_SIGN_IN_USER_ERROR:
      return {
        ...state,
        validationError: action.payload
      };
    case ActionType.RESET_ERRORS:
      return {
        ...state,
        validationError: action.payload
      };
    case ActionType.LOGOUT_USER:
      return initialState;
  }
  return state;
};

const transformUserData = (data) => {
  const {avatar_url: avatarUrl, ...other} = data;
  return {avatarUrl, ...other};
};

const Operation = {
  signInUser: (params) => (dispatch, getState, api) => {
    return api
      .post(`/login`, params)
      .then((res) => {
        const userInfo = transformUserData(res.data);
        dispatch(ActionCreator.signInUser(userInfo));
      }).catch((error) => {
        dispatch(ActionCreator.signInUserError(error.message));
      });
  },
  checkUser: () => (dispatch, getState, api) => {
    return api
      .get(`/login`)
      .then((res) => {
        const userInfo = transformUserData(res.data);
        dispatch(ActionCreator.signInUser(userInfo));
      }).catch((error) => {
        dispatch(ActionCreator.logoutUser());
        dispatch(ActionCreator.signInUserError(error.message));
      });
  }
};

export {
  Operation,
  reducer,
  transformUserData
};
