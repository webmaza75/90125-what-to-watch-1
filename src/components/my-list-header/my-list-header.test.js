import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';

import MyListHeader from './my-list-header.jsx';
import user from '../../mocks/user.js';

const context = {};

describe(`MyListHeader correctly renders`, () => {
  it(`MyListHeader correctly renders for authorized user`, () => {
    const myListHeader = renderer
      .create(<StaticRouter location="someLocation" context={context}>
        <MyListHeader
          user={user}
        />
      </StaticRouter>)
      .toJSON();

    expect(myListHeader).toMatchSnapshot();
  });
});
