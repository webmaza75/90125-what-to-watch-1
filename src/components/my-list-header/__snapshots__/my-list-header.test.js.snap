// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListHeader correctly renders MyListHeader correctly renders for authorized user 1`] = `
<header
  className="page-header user-page__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link "
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    My list
  </h1>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="https://es31-server.appspot.comimg/1.png"
        width="63"
      />
    </div>
  </div>
</header>
`;
